From 1ff90eb9e54ee23ed86c5497cd841f3040157bce Mon Sep 17 00:00:00 2001
From: Kiran Rudramuni <kchitrik@codeaurora.org>
Date: Wed, 2 Sep 2015 11:00:30 -0700
Subject: [PATCH] libhardware-legacy: Add mechanism for applying properties
 values

After wigig driver is loaded, the code triggers service registered
at init rc script. The service starts bash script which in turn
alters the configuration of network performance related parameters
such as tcp buffers size, mtu size, and rps

Change-Id: If28cb6d9a129bfea4398f1a9090d64491dad4932
---
 wifi/wifi.c | 34 ++--------------------------------
 1 file changed, 2 insertions(+), 32 deletions(-)

diff --git a/wifi/wifi.c b/wifi/wifi.c
index d3f545f..8904646 100644
--- a/wifi/wifi.c
+++ b/wifi/wifi.c
@@ -235,37 +235,6 @@ int is_wifi_driver_loaded() {
 #endif
 }
 
-void set_mtu() {
-    char interface_name[PROPERTY_VALUE_MAX];
-    char mtu_str_value[PROPERTY_VALUE_MAX];
-    int mtu_value = MTU_DEFAULT_VALUE;
-    int ret;
-    struct ifreq ifr;
-
-    property_get(WLAN_IF_PROPERTY, interface_name, "wlan0");
-    property_get(MTU_VALUE_PROPERTY, mtu_str_value, NULL);
-
-    sscanf(mtu_str_value, "%d", &mtu_value);//ignoring errors, there is default
-
-    /* init socket */
-    int ifc_ctl_sock = socket(AF_INET, SOCK_DGRAM, 0);
-    if (ifc_ctl_sock < 0) {
-        ALOGE("Cannot create raw socket. Error:%s", strerror(errno));
-        return;
-    }
-
-    /* set up interface name for request */
-    strncpy(ifr.ifr_name, interface_name, IFNAMSIZ);
-    ifr.ifr_name[IFNAMSIZ - 1] = 0;
-    ifr.ifr_mtu = mtu_value;
-    /* go */
-    ret = ioctl(ifc_ctl_sock, SIOCSIFMTU, &ifr);
-    if (ret < 0) {
-        ALOGE("Error writing MTU=%d with IOCTL. Error:%s", mtu_value, strerror(errno));
-    }
-    close(ifc_ctl_sock);
-}
-
 int wifi_load_driver()
 {
 #ifdef WIFI_DRIVER_MODULE_PATH
@@ -290,7 +259,8 @@ int wifi_load_driver()
     while (count-- > 0) {
         if (property_get(DRIVER_PROP_NAME, driver_status, NULL)) {
             if (strcmp(driver_status, "ok") == 0) {
-                set_mtu();
+                property_set("ctl.start", "wigignetcfg");
+                sched_yield();
                 return 0;
             } else if (strcmp(driver_status, "failed") == 0) {
                 wifi_unload_driver();
-- 
1.8.2.1

